### ISSUE TITLE
Hints for Enum names and column names based on Levenshtein distance.

### ISSUE THREAD

**Issue by alexey-milovidov** [ClickHouse/ClickHouse#17112]
**Use case**
Provide "Did you mean: ..." with nearest matches when user made a typo in Enum name or column name.

**Describe the solution you'd like**
See `NamePrompter.h`

**Comment by fastio** [comment]
It looks that the column name should be registered in the IHints.
Please assign it to me.

**Comment by alexey-milovidov** [comment]
Yes, looks like we only have to put every name into IHints and then ask for possible matches.

**Comment by alexey-milovidov** [comment]
Hints for Enum names are still to be implemented. A test:
```
SELECT CAST('Helo' AS Enum('Hello' = 1, 'World' = 2))
```
